homeassistant/components/smlight/config_flow.py

```diff
@@ -0,0 +1,150 @@
+"""Config flow for SMLIGHT Zigbee integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from pysmlight.exceptions import SmlightAuthError, SmlightConnectionError
+from pysmlight.web import Api2
+import voluptuous as vol
+
+from homeassistant.components import zeroconf
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import DOMAIN
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+STEP_AUTH_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class SmlightConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for SMLIGHT Zigbee."""
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.client: Api2
+        self.host: str | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self.host = user_input[CONF_HOST]
+            assert self.host is not None
```

Comment:
this was because mypy complaining about type of `self.host` being optional, but param to Api2 is mandatory. I changed to use an intermediate host variable instead, hopefully that is clearer?